win32ss/user/ntuser/kbdlayout.c

```diff
@@ -871,10 +838,10 @@ co_IntUnloadKeyboardLayoutEx(
 
     UserDerefObjectCo(pKL); /* Release reference */
 
-    if (pti->pDeskInfo->fsHooks)
+    if (ISITHOOKED(WH_SHELL))
```

Comment:
What happens if ghKLSentToShell is already NULL but this if-block is taken? Is it still to send the notification?

Reply:
Unloading will activate the next layout, so a notification of language zero never happen.